### DISCUSSION TITLE
Replacing state with a different slice does not work

### DISCUSSION THREAD

**Discussion by chris-dem** [reduxjs/redux-toolkit#4542] [Q&A] [answered]
I am using a reducer path `test`. Depeding on the url path, I want to inject a different state. Issue is that, after my first slice is injected, I am not seeing any updates to my state, even after enabling the `overrideExisting` flag. Am I missing something?

**Comment by markerikson** [comment]
I have no idea what you're trying to do here.  There's no code, and I don't know what you mean by "replace state" vs "inject a different state".

**Reply by chris-dem** [reply]
https://medium.com/@matheusmm/redux-injecting-reducers-dynamically-e4e193eaa905
mainly referring to this part of the document here

**Reply by chris-dem** [reply]
What I am trying to do is as follows: This is my main reducer:
```ts
export const appReducer = combineSlices({
  ...
}).withLazyLoadedSlices<LazyLoadedSlices>();
```
My `LazyLoadedState`
```ts
export interface LazyLoadedSlices {
  test: countAState | countBState | null
}
```
And using react router, depending on the route, I am using one of the two to load the state
```ts
export const lazyLoader : LoaderFunction<any> = (_args) => {
  countASlice.injectInto(appReducer,{ reducerPath: "test", overrideExisting: true })
  return {}
}

export const lazyLoader1 : LoaderFunction<any> = (_args) => {
  countBSlice.injectInto(appReducer,{ reducerPath: "test", overrideExisting: true })
  return {}
}
```

**Reply by chris-dem** [reply]
What happens is that,  if we run the `lazyLoader` first, redux devtools show the `countASlice`. If we try to use the second loader afterwards, the state stays the same. Only difference is when using an empty reducer, but even after that, it doesnt transition to any othe state

**Reply by EskiMojo14** [reply]
replacing the reducer doesn't replace the state - you need to dispatch an action for the state to change.

**Reply by chris-dem** [reply]
That still doesn't work, tried toggling an action but the state stayed the same.

**Reply by EskiMojo14** [reply]
any chance you could put together a minimal reproduction of this?

**Reply by chris-dem** [reply]
Can you try with this sandbox? https://codesandbox.io/p/sandbox/redux-toolkit-playground-forked-r58wj5

**Reply by EskiMojo14** [reply, marked as answer]
it's called *initialState* for a reason - it's only used when the reducer is called with `undefined`. In this case, we already have state for this reducer, so it's using that state instead and returning it.
```ts
let reducer1 = (state = { id: 1 }, action) => state;

let reducer2 = (state = { id: 2 }, action) => state;

let state = reducer1(undefined, { type: "INIT" });

state = reducer2(state, { type: "REPLACE" });

// state is still { id: 1 }
```

**Reply by chris-dem** [reply]
So what would you recommend to do if I want to replace the state using the original state. And also what is the point of the `overrideExisting`?

**Reply by EskiMojo14** [reply]
`overrideExisting` allows replacing the reducer, which is entirely different from replacing the state.

At a guess, I suspect you would need to wrap your slice reducer in a function that detects if the state is from the other reducer and replaces it before the slice reducer handles it.

Overall this isn't something we'd consider a common use case - we'd expect a given reducer in a given location to always handle the same data.

**Reply by chris-dem** [reply]
I see thank you
